server-spi-private/src/main/java/org/keycloak/requestfilter/IpAccessFilterProviderFactory.java

```diff
@@ -0,0 +1,71 @@
+package org.keycloak.requestfilter;
+
+import org.jboss.resteasy.spi.ResteasyProviderFactory;
+import org.keycloak.Config;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.KeycloakSessionFactory;
+import org.keycloak.provider.ProviderConfigProperty;
+import org.keycloak.provider.ProviderConfigurationBuilder;
+
+import java.util.List;
+import java.util.Optional;
+
+public class IpAccessFilterProviderFactory implements RequestFilterProviderFactory {
+
+    public static final String PROVIDER_ID = "ip-access";
+
+    @Override
+    public IpAccessFilterProvider create(KeycloakSession session) {
+        return new IpAccessFilterProvider();
+    }
+
+    private boolean isEnabled = false;
+
+    @Override
+    public void init(Config.Scope config) {
+        this.isEnabled = Optional.ofNullable(config.getBoolean("enabled")).orElse(false);
+    }
+
+    @Override
+    public void postInit(KeycloakSessionFactory factory) {
+        if (isEnabled) {
+            ResteasyProviderFactory.getInstance().getContainerRequestFilterRegistry()
```

Comment:
Ideally, we should do the opposite. Obtain the provider from a platform-specific filter/interceptor and then delegate validation/authorization to the provider.

Custom extensions should not rely on the stack we are running. Recently, we did changes to reduce as much as possible the dependency on Resteasy APIs but provide our own API to abstract the underlying stack. For instance, the `RequestFilter` provider should be accepting `org.keycloak.http.HttpRequest` instances to do their job.